src/Login/ProtectedContent.jsx

```diff
@@ -53,13 +53,16 @@ class ProtectedContent extends React.Component {
             store.dispatch({ type: 'CLEAR_QUERY_NODES' }),
           ],
         )
+          .then(() => this.checkLoginStatus(store, this.state)
+            .then((newState) => ((this.props.public) ? { ...newState, redirectTo: null } : this.checkQuizStatus(newState))) // don't redirect for public pages
+            .then((newState) => ((this.props.public) ? { ...newState, redirectTo: null } : this.checkApiToken(store, newState))))
           .then(
-            () => this.checkLoginStatus(store, this.state)
-              .then((newState) => ((this.props.public) ? { ...newState, redirectTo: null } : this.checkQuizStatus(newState))) // don't redirect for public pages
-              .then((newState) => ((this.props.public) ? { ...newState, redirectTo: null } : this.checkApiToken(store, newState))),
-          ).then(
             (newState) => this.checkUseWarning(store, newState), // check for existence of cookie to popup Use Warning
           )
+          .then(
+            // check for site access if property set
+            (newState) => this.checkSiteAccess(newState),
```

Comment:
might better check for `userAccessToSite?.enabled` here so we don't need always call `this.checkSiteAccess()`

Reply:
And this feature doesn't really works for public pages right? If a page is public, it suppose to be accessible for all users. But if this feature is turned on, it will block user's access?

Reply:
this feature is to block access to all pages, there is no feature doc as they want it live by Friday. This PR is not to merge it to master but instead to a separate release branch for VA. i will get to your performance improvements

Reply:
@mfshao i think the logic should stay in the this.checkSiteAccess function, but i will refactor that function so it does not do any additional work if the feature is not enabled